src/redux/reducers/movieReducer.js

```diff
@@ -0,0 +1,86 @@
+import {
+    FETCH_MOVIES_BEGIN,
+    FETCH_MOVIES_SUCCESS,
+    FETCH_MOVIES_FAILURE,
+    FETCH_FILMID_BEGIN,
+    FETCH_FILMID_SUCCESS,
+    FETCH_FILMID_FAILURE,
+    SORT_RELEASE,
+    SORT_RATING
+} from '../actions/moviesActions';
+
+const initialState = {
+    movies: { data: [], total: 0 },
+    filmId: {},
+    loading: false,
+    error: null
+};
+
+export default function movieReducer(state = initialState, action) {
+    switch(action.type) {
+        case FETCH_MOVIES_BEGIN:
+            return {
+                ...state,
+                loading: true,
+                error: null
+            };
+
+        case FETCH_MOVIES_SUCCESS:
+            return {
+                ...state,
+                loading: false,
+                movies: action.payload.data
+            };
+
+        case FETCH_MOVIES_FAILURE:
+            return {
+                ...state,
+                loading: false,
+                error: action.payload.error,
+                movies: { data: [], total: 0 },
+            };
+
+        case FETCH_FILMID_BEGIN:
+            return {
+                ...state,
+                loading: true,
+                error: null
+            };
+
+        case FETCH_FILMID_SUCCESS:
+            return {
+                ...state,
+                loading: false,
+                filmId: action.payload
+            };
+
+        case FETCH_FILMID_FAILURE:
+            return {
+                ...state,
+                loading: false,
+                error: action.payload.error,
+                filmId: {},
+            };
+
+        case SORT_RELEASE:
+            let sortDataByRelease = state.movies.data.slice().sort((a, b) => new Date(b.release_date) - new Date(a.release_date));

```

Comment:
while chaining use please every method on the new line in so called columns. It can be very useful while code reviewing